10x/count/count.wdl

```diff
@@ -0,0 +1,1427 @@
+    Array[String] bam_comments = read_lines("bam_comments.lines")
+  }
+}
+
+#stage ALIGN_READS(
+#    in  fastq[]  reads,
+#    in  fastq[]  read2s,
+#    in  string[] read_groups,
+#    in  path     reference_path,
+#    in  int      threads,
+#    in  int      max_hits_per_read,
+#    out bam[]    genome_output,
+#    out bool     paired_end,
+#    src py       "stages/counter/align_reads",
+#) split using (
+#    in  fastq    read_chunk,
+#    in  fastq    read2_chunk,
+#    in  string[] read_group,
+#)
+task align_reads_split {
+  # The split phase doesn't really need the reads files, so it's fine just giving it strings
+  # that are (invalid) paths. It's just splitting the list up.
+  Array[String] read_paths
+  Array[String] read_groups
+  File reference_path
+  
+  command <<<
+
+    # Create the args
+    echo '{"threads": 2}' > _args
+    read_paths_json='["${sep='","' read_paths}"]'
+    echo "$(jq --arg var "$read_paths_json" '. + {"reads": ( $var | fromjson )}' _args)" > _args
+    read_groups_json='["${sep='","' read_groups}"]'
+    echo "$(jq --arg var "$read_groups_json" '. + {"read_groups": ( $var | fromjson )}' _args)" > _args
+    echo "$(jq --arg var "[]" '. + {"read2s": ( $var | fromjson )}' _args)" > _args
+    echo "$(jq --arg var "${reference_path}" '. + {"reference_path": $var}' _args)" > _args
+    
+    # Run martian
+    mv /_jobinfo .  
+    stage_path=/cellranger/mro/stages/counter/align_reads
+    stage_phase=split
+    /martian/adapters/python/martian_shell.py $stage_path $stage_phase . . run_file_whatever
+
+    # Break the chunks into separate files so we can scatter over them
+    jq -c '.chunks[]' _stage_defs | split -l 1 - chunk_
+  >>>
+ 
+  runtime {
+    docker: "marcusczi/cellranger_clean:cromwell"
+  }
+
+  output {
+    File stage_defs = "_stage_defs"
+    Array[File] chunks = glob("chunk_*")
+  }
+}
+
+task align_reads_main {
+
+  Int max_hits_per_read
+  File reference_path
+  # A fastq produced by extract_reads 
+  File chunked_fastq
+  # The chunk definition produced by align_reads_split
+  File chunk
+
+  command <<<
+    
+    # Create args
+    threads=$(nproc)
+    echo '{"threads": '$threads', "read2_chunk": null}'  > _args
+    echo "$(jq --arg var ${reference_path} '. + {"reference_path": $var}' _args)" > _args
+    echo "$(jq --arg var ${max_hits_per_read} '. + {"max_hits_per_read": ( $var | fromjson )}' _args)" > _args
+    echo "$(jq --arg var "$(<${chunk})" '. |=  .+ ($var | fromjson) ' _args)" > _args
+    echo "$(jq --arg var "${chunked_fastq}" '. + {"read_chunk": $var}' _args)" > _args
+    
+    # Create outs
+    echo '{"genome_output": "genome_output.bam"}' > _outs
+
+    mv /_jobinfo .  
+    stage_path=/cellranger/mro/stages/counter/align_reads
+    stage_phase=main
+    /martian/adapters/python/martian_shell.py $stage_path $stage_phase . . run_file_whatever
+  >>>
+
+  runtime {
+    docker: "marcusczi/cellranger_clean:cromwell"
```

Comment:
I noticed that you did not specific memory requirement in runtime. is it okay? just curious....

Reply:
It's OK for local runs (local cromwell warns you that it's ignoring both disks and memory when you run it with them specified).

Reply:
good to know, thanks!